### PR TITLE
Tw 1492 ads replacement rework for mobile browsing

### PR DESCRIPTION
**Pull request by keshan3262**

### CODE REVIEW

File: src/utils/schemas.ts [hunk 1]
```diff
@@ -173,9 +173,35 @@ const permanentAdPlacesRulesSchema = arraySchema()
...
+            }
+
+            throw this.createError({ path: this.path, message: 'Both `width` and `height` fields must be specified' });
+          })
+          .default(undefined) as unknown as IObjectSchema<{ width: string; height: string } | undefined>,
         stylesOverrides: arraySchema().of(adStylesOverridesSchema.clone().required()),
         shouldHideOriginal: booleanSchema(),
-        extVersion: versionRangeSchema.clone().required()
+        extVersion: versionRangeSchema.clone().required(),
+        displayWidth: versionRangeSchema.clone().test('valid-boundary-values', (value: string | undefined) => {
```
[the remark this answers is not in the document]
Comment (keshan3262) [reply src/utils/schemas.ts:193]: You may be wondered but it is, see https://github.com/madfish-solutions/templewallet-backend/pull/167/files#diff-db5cfd5f3b986278d291b7ef7f8b55006f22aa79495a1636037690810f7587dfR306

**Review by lendihop — APPROVED** [review]

**Review by lendihop — APPROVED** [review]

**Review by IrynaKhyzhynska — APPROVED** [review]
done
